src/js/program/media-element-pool.ts

```diff
@@ -20,30 +36,30 @@ export default function MediaElementPool() {
     let primed = false;
 
     return {
-        primed() {
+        primed(): boolean {
             return primed;
         },
-        prime() {
+        prime(): void {
             elements.forEach(primeMediaElementForPlayback);
             primed = true;
         },
-        played() {
+        played(): void {
             primed = true;
         },
-        getPrimedElement() {
+        getPrimedElement(): OptionalUndefVideo | null {
             if (pool.length) {
                 // Shift over pop so that we cycle through elements instead of reusing the same one
                 return pool.shift();
```

Comment:
This would never be undefined

Reply:
interestingly TS does not recognize the `.length` check as a safeguard here - so it's expecting me to cover the case where `shift` returns a non-element (length 0). i have an idea of how to change the func to fix.